opdatasigverify.md

```diff
@@ -0,0 +1,122 @@
+# Enable oracle-based data import via OP_DATASIGVERIFY
+
+Version 0.2, 2018-02-22
+
+## Introduction
+
+OP_DATASIGVERIFY allows signed data to be imported into a script.  This data can then have many uses, depending on the rest of the script, such as deciding spendability of several possible addresses.  This opcode therefore enables the powerful blockchain concept of an "oracle" -- an entity that publishes authoritative statements about extra-blockchain events -- to be used in the Bitcoin Cash blockchain.  For an example use of how this opcode can be used to enable binary contracts on any security or betting on any quantitatively decidable event (such as a sports match) please [click here](https://medium.com/@g.andrew.stone/bitcoin-scripting-applications-decision-based-spending-8e7b93d7bdb9).  But this is just one example; as the Bitcoin Cash Script language grows in expressiveness, it is anticipated that this opcode will be used in many other applications.
+
+## Requirements on miners, full nodes, and clients
+
+Miners, full nodes, and clients will implement the OP_DATASIGVERIFY opcode and activate it during the May 2018 hard fork.
+
+## OP_DATASIGVERIFY Specification
+
+*OP_DATASIGVERIFY uses a new opcode number*
+
+    Opcode (decimal): 187
+    Opcode (hex): 0xbb
+
+### Implementation details:
+When OP_DATASIGVERIFY is executed, the stack should look like:
+
+*top of stack*
+* *pubkeyhash*
+* *type and signature*
+* *data*
+
+If there are less then 3 items on the stack, the script fails.  If the pubkeyhash field is not 20 bytes, the script fails.  If the *'type and signature'* field is not 66 bytes, the script fails.  If the last byte of the *'type and signature'* field is not DATASIG_COMPACT_ECDSA (1), the script fails.
+
+The last byte of the *'type and signature'* (stack top - 1) field defines the signature type, and previous bytes are the actual signature.  Note that this format is different from the OP_CHECKSIG format -- its SIGHASH flag byte has no relationship to this signature type byte.
```

Comment:
Why switch the convention that OP_CHECKSIG uses?

Reply:
Does that even make sense?  The flag here wouldn't be with respect to which part of the transaction to sign.

Reply:
the op_checksig sighashtype bits don't make any sense in the context of datasigverify we need to change the convention.